components/kyma-environment-broker/internal/runtime/converter.go

```diff
@@ -167,4 +174,48 @@ func (c *converter) ApplyUpdateOperations(dto *pkg.RuntimeDTO, oprs []internal.U
 		c.applyOperation(&o.Operation, &op)
 		dto.Status.Update.Data = append(dto.Status.Update.Data, op)
 	}
+	c.adjustRuntimeState(dto)
+}
+
+func (c *converter) adjustRuntimeState(dto *pkg.RuntimeDTO) {
+	lastOp := dto.LastOperation()
+	switch lastOp.State {
+	case string(domain.Succeeded):
+		dto.Status.State = pkg.StateSucceeded
+		if lastOp.Type == pkg.Suspension {
+			dto.Status.State = pkg.StateSuspended
+		}
+	case string(domain.Failed):
+		dto.Status.State = pkg.StateFailed
+		switch lastOp.Type {
+		case pkg.UpgradeKyma, pkg.UpgradeCluster, pkg.Update:
+			dto.Status.State = pkg.StateError
+		}
+	case string(domain.InProgress):
+		switch lastOp.Type {
+		case pkg.Provision, pkg.Unsuspension:
+			dto.Status.State = pkg.StateProvisioning
+		case pkg.Deprovision, pkg.Suspension:
+			dto.Status.State = pkg.StateDeprovisioning
+		case pkg.UpgradeKyma, pkg.UpgradeCluster:
+			dto.Status.State = pkg.StateUpgrading
+		case pkg.Update:
+			dto.Status.State = pkg.StateUpdating
+		}
+	default:
+		dto.Status.State = pkg.StateSucceeded
+	}
+
+	if dto.Status.Suspension != nil && dto.Status.Suspension.Count > 0 {
+		// there is no unsuspension operation or the suspension is started after last unsuspension
+		if dto.Status.Unsuspension == nil ||
+			(dto.Status.Unsuspension.Count > 0 && dto.Status.Unsuspension.Data[0].CreatedAt.Before(dto.Status.Suspension.Data[0].CreatedAt)) {
```

Comment:
double check if Data[0] has newest entries

Reply:
listing operations in the DB is ordered by date